### ISSUE TITLE
Special Functions and Logic Switches cross transfer

### ISSUE THREAD

**Issue by ninodesposati** [FrSkyRC/ETHOS-Feedback-Community#3880]
When I build a new model many times I use a lot or all of the other special functions and Logic switches I created in other models. I'd like to see a way to transfer all special functions and logic switches to other models. Thanks

**Comment by bsongis-frsky** [comment]
I see, it should be possible
- Choose a model
- Show everything which could be imported
- Let the user choose
- Do the import job

**Comment by ninodesposati** [comment]
Perfect!! Thanks

**Comment by Alex00101** [comment]
May I please post a suggestion or consideration?  Perhaps the same outcome plus more could be achieved by considering some sort of template system (perhaps using excellent folder/clone model functionality plus:
- settings transfer (as per previous posts here)
- extended model description (based on md file with better scrolling etc)
- inclusion on create model template 
- ...

I'm sure it has been suggested before

**Comment by sebakke** [comment]
I like the idea of selective cloning of a model where you can take parts as described above.  So that's one use case for bringing a common setup from another plane.  However, there are cases where I'd like to be able to copy things from one model to another later on when the destination model already exists.  Would it be possible to add a copy/import feature for logic switches, special functions, and mixes?   That would save a lot of time rather than having to re-implement something that you've perfected on another model.

**Comment by spoke2570** [comment]
I love this, in general. 

I would also like to see some copy/paste or send function so SFs, LSs, curves, mixes, timers and etc. can be brought into a model without loading the model with the item wanted, find the item, copy it, load the model being worked on go to the place the copied item belongs and paste it. Its can be done this way now, but it is usually more cumbersome than rebuilding from scratch. 

Maybe the UI could look like a tile in the new mix window for an 'imported mix', and a similar line in the actions menu that drops when the enter button is pressed. For LS and SF when the '+' is pressed show an option menu for new item or imported item. 

Maybe a whole function where we can select multiple LS, SF, curve, etc from another model via a check box system for import to a model that is being worked on. 

Currently when I do this the risk of making a critical oops while copying an item from another model is high. I open the model and use a fiddly menu for copying the thing I want while trying to not accidentally delete or modify it. This solution would make those things accessible to the new model being worked on and protect the models they are being lifted from.

**Comment by hrenz** [comment]
need a Companion for this too

**Comment by EvPetrov** [comment]
The PC "Companion" is badly needed.

**Comment by pstasek** [comment]
A human-readable JSON, XML, or other model format (at least for Ethos Suite as a compiler to the bin format) would probably be a good foundation to start implementing the Ethos Suite editor and that would also allow copy/pasting of certain features between models.

Such a snippet might be also possible to be imported to a model using file Browser.

This would also allow people to share widely usable features instead of explaining how to program certain stuff.

What do you think?

**Comment by ninodesposati** [comment]
I am happy to see that there are many people like myself who want the same feature. Hopefully this can get implemented soon.

**Comment by Richrfl** [comment]
I fully agree with the idea.

**Comment by spoke2570** [comment]
From Bertrand on thread #1440 "> Planned for 2024 after Model Editor inside Ethos Suite is done (this is the BIG missing feature in Ethos if you compare OpenTX with Ethos)"

I have faith....

**Comment by Toffs** [comment]
I believe others have recently asked for a copy/paste for Vars also (between models).

**Comment by spoke2570** [comment]
Yes. Outputs, timers, trims, mixes curves, vars, LS, SF.. I believe they all should have an import function right in the clone-copy-edit list so one doesn’t have to leave a current model to get a wanted component from another model.

**Comment by ninodesposati** [comment]
Where are we with this? With so many updates special functions and logic switches don't make it over to the model and needs to be recreated.

**Comment by JustPlaneChris** [comment]
I'd just like to add my vote to a "companion-like" addition to the Ethos Suite to allow easy model editing and copying logic switches and such from one model to another. I very much miss that ability from EdgeTX.

**Comment by Toffs** [comment]
> Yes. Outputs, timers, trims, mixes curves, vars, LS, SF.. I believe they all should have an import function right in the clone-copy-edit list so one doesn’t have to leave a current model to get a wanted component from another model.

It’s been a little while, any news on something like the above becoming a reality?

Whilst cloning a plane / having a template is great for a ‘new’ plane, it doesn’t work when you’re doing new programming or features and then need to apply to multiple/all existing models.

If we rewrite a whole new Vars program, say with 5 entries, or a new curve mix - we then might need to manually make that for 5, 10 (or 25 if you’re super keen) model files. Painful :(
Copy/Paste is the answer, even better if it was part of a new computer companion app :D
